src/olmo_core/train/callbacks/gap_monitor.py

```diff
@@ -150,10 +155,19 @@ def record_tensor_stats(
                     merge_strategy=MetricMergeStrategy.sum,
                 )
         else:
-            max_ = get_local_tensor(tensor).abs().max()
             var, mean = var_mean(tensor)
+            local_tensor = get_local_tensor(tensor)
+            if local_tensor.numel() > 0:
+                local_max = local_tensor.abs().max()
+            else:
+                # Use 0.0 as sentinel value for empty tensors in max reduction.
+                # Since we're taking abs(), all actual values are >= 0, so 0.0
+                # won't affect the max reduction when other processes have non-empty tensors.
+                local_max = torch.tensor(0.0, device=tensor.device, dtype=tensor.dtype)
```

Comment:
This way would avoid the host-device sync:
```suggestion
                local_max = torch.zeros([], device=tensor.device, dtype=tensor.dtype)
```